plugins/decklink/decklink-source.cpp

```diff
@@ -70,8 +70,11 @@ static void decklink_update(void *data, obs_data_t *settings)
 	decklink->SetColorSpace(colorSpace);
 	decklink->SetColorRange(colorRange);
 	decklink->SetChannelFormat(channelFormat);
-	decklink->Activate(device, id);
 	decklink->hash = std::string(hash);
+	decklink->swap = obs_data_get_bool(settings, SWAP);
+	if (decklink->swap)
+		decklink->Deactivate();
```

Comment:
You can actually check if the swap value is different in update here:
https://github.com/obsproject/obs-studio/blob/master/plugins/decklink/DecklinkInput.cpp#L60

and then the following code will stop the capture and restart the capture.

Reply:
done